internal/gitrepo/gitrepo_test.go

```diff
@@ -665,3 +649,353 @@ func TestGetDir(t *testing.T) {
+			if (err != nil) != test.wantErr {
+				t.Errorf("ChangedFilesInCommit() error = %v, wantErr %v", err, test.wantErr)
+				return
+			}
+			if diff := cmp.Diff(test.wantFiles, gotFiles); diff != "" {
+				t.Errorf("ChangedFilesInCommit() mismatch (-want +got):\n%s", diff)
+			}
+		})
+	}
+}
+
+// setupRepoForGetCommitsTest creates a repository with a few commits and tags.
+func setupRepoForGetCommitsTest(t *testing.T) (*LocalRepository, map[string]string) {
+	t.Helper()
+	repo, dir := initTestRepo(t)
+	commits := make(map[string]string)
+
+	// Commit 1
+	commit1 := createAndCommit(t, repo, "file1.txt", []byte("content1"), "feat: commit 1")
+	commits["commit1"] = commit1.Hash.String()
+
+	// Tag for commit 1
+	if _, err := repo.CreateTag("v1.0.0", commit1.Hash, nil); err != nil {
+		t.Fatalf("CreateTag failed: %v", err)
+	}
+
+	// Commit 2
+	commit2 := createAndCommit(t, repo, "file2.txt", []byte("content2"), "feat: commit 2")
+	commits["commit2"] = commit2.Hash.String()
+
+	// Commit 3
+	commit3 := createAndCommit(t, repo, "file3.txt", []byte("content3"), "feat: commit 3")
+	commits["commit3"] = commit3.Hash.String()
+
+	return &LocalRepository{Dir: dir, repo: repo}, commits
+}
+
+func TestGetCommitsForPaths(t *testing.T) {
+	t.Parallel()
+
+	repo, commits := setupRepoForGetCommitsTest(t)
+
+	for _, test := range []struct {
+		name                string
+		paths               []string
+		tagName             string
+		sinceCommit         string
+		wantCommits         []string
+		wantErr             bool
+		useGetCommitsForTag bool
+	}{
+		{
+			name:        "GetCommitsForPathsSinceCommit - one path, one commit",
+			paths:       []string{"file2.txt"},
+			sinceCommit: commits["commit1"],
+			wantCommits: []string{"feat: commit 2"},
+		},
+		{
+			name:        "GetCommitsForPathsSinceCommit - all paths, all commits",
+			paths:       []string{"file1.txt", "file2.txt", "file3.txt"},
+			sinceCommit: "",
+			// The current implementation skips the initial commit.
+			wantCommits: []string{"feat: commit 3", "feat: commit 2"},
+		},
+		{
+			name:        "GetCommitsForPathsSinceCommit - no matching commits",
+			paths:       []string{"non-existent-file.txt"},
+			sinceCommit: "",
+			wantCommits: []string{},
+		},
+		{
+			name:                "GetCommitsForPathsSinceTag - all paths, multiple commits",
+			paths:               []string{"file2.txt", "file3.txt"},
+			tagName:             "v1.0.0",
+			wantCommits:         []string{"feat: commit 3", "feat: commit 2"},
+			useGetCommitsForTag: true,
+		},
+		{
+			name:                "GetCommitsForPathsSinceTag - invalid tag",
+			paths:               []string{"file2.txt"},
+			tagName:             "non-existent-tag",
+			wantCommits:         []string{},
+			wantErr:             true,
+			useGetCommitsForTag: true,
+		},
+	} {
+
+		t.Run(test.name, func(t *testing.T) {
+			var (
+				gotCommits []*Commit
+				err        error
+			)
+			if test.useGetCommitsForTag {
```

Comment:
I think it's better to create a separate test for `GetCommitsForPathsSinceTag`.

Reply:
done.